diff -ru protobuf-2.0.2.orig/src/google/protobuf/compiler/command_line_interface.cc protobuf-2.0.2/src/google/protobuf/compiler/command_line_interface.cc
--- protobuf-2.0.2.orig/src/google/protobuf/compiler/command_line_interface.cc	2009-03-01 10:14:52.000000000 +0000
+++ protobuf-2.0.2/src/google/protobuf/compiler/command_line_interface.cc	2009-03-01 10:27:11.000000000 +0000
@@ -44,6 +44,7 @@
 #include <errno.h>
 #include <iostream>
 #include <ctype.h>
+#include <stdio.h>
 
 #include <google/protobuf/compiler/command_line_interface.h>
 #include <google/protobuf/compiler/importer.h>
diff -ru protobuf-2.0.2.orig/src/google/protobuf/stubs/strutil.cc protobuf-2.0.2/src/google/protobuf/stubs/strutil.cc
--- protobuf-2.0.2.orig/src/google/protobuf/stubs/strutil.cc	2009-03-01 10:14:52.000000000 +0000
+++ protobuf-2.0.2/src/google/protobuf/stubs/strutil.cc	2009-03-01 10:15:40.000000000 +0000
@@ -35,6 +35,7 @@
 #include <float.h>    // FLT_DIG and DBL_DIG
 #include <limits>
 #include <limits.h>
+#include <stdio.h>
 
 #ifdef _WIN32
 // MSVC has only _snprintf, not snprintf.
diff -ru protobuf-2.0.2.orig/src/google/protobuf/text_format.cc protobuf-2.0.2/src/google/protobuf/text_format.cc
--- protobuf-2.0.2.orig/src/google/protobuf/text_format.cc	2009-03-01 10:14:52.000000000 +0000
+++ protobuf-2.0.2/src/google/protobuf/text_format.cc	2009-03-01 10:23:07.000000000 +0000
@@ -36,6 +36,7 @@
 #include <math.h>
 #include <stack>
 #include <limits>
+#include <stdio.h>
 
 #include <google/protobuf/text_format.h>
 
